apps/omg_watcher/lib/block_getter/core.ex

```diff
@@ -283,35 +270,29 @@ defmodule OMG.Watcher.BlockGetter.Core do
         ) ::
           {:ok | {:needs_stopping, block_error()}, %__MODULE__{},
            [] | list(Event.InvalidBlock.t()) | list(Event.BlockWithholding.t())}
-          | {:error, :duplicate | :unexpected_blok}
+          | {:error, :duplicate | :unexpected_block}
   def handle_downloaded_block(
-        %__MODULE__{number_of_blocks_being_downloaded: number_of_blocks_being_downloaded} = state,
-        response
+        %__MODULE__{
```

Comment:
it seems like you're reverting a previous bug fix. If a withholding is detected you should decrease the number of blocks being downloaded (i.e. a slot is free now).